### DISCUSSION TITLE
[Hackathon Use Case] Implement features of CDM Collateral model to meet common Collateral Working Group use cases

### DISCUSSION THREAD

**Discussion by iansloyan** [finos/finos-pmcs#251] [Ideas]
# Implement features of CDM Collateral model to meet common Collateral Working Group use cases
_From @iansloyan (ISDA)_

## Business Problem
There are 2 suggested use cases for a hacker to download the CDM model and focus on the collateral model specifically and use it to demonstrate the following use cases - additional credit could be given to using FDC3 and/or Morphir to exchange and/or run the basic tests on the data objects. 

1. Using the CDM collateral representation already built, build functionality to test collateral eligibility against multiple jurisdictions’ minimum collateral requirements and specific eligible collateral schedules.
_Essentially this is testing different criteria sets built from the EligibleCollateralSchedule root type against each other - e.g., the regulation* says you must at a minimum provide collateral of bonds of 5 years or more from OECD countries vs counterparty says you can only post bonds of from France or Germany which are 7 years and rated AA+ vs internal inventory rule which says can post any bonds you like as long as 5 years or more maturity. The intersection of these criteria sets becomes the effective criteria you need to apply when picking a bond to post to counterparty._ 
*https://www.isda.org/a/EqxgE/Eligible-Collateral-Comparison-010523.pdf 

2. Using the CDM collateral representation already built, build functionality to compare new and previous eligible collateral schedules and identify changes to be ingested into a collateral management system. Develop a process to run this look, find, change function daily.
_This is a different dimension to the above where testing against changing criteria on a daily or regular basis and has many use cases in collateral management._

## Expected outcome

JSON files are provided to represent specific criteria for types of bonds that are eligible as collateral under certain regulatory margin rules. 
[EligibleCollateralCriteria_1_28-04-2023 2.zip](https://github.com/finos/community/files/11383291/EligibleCollateralCriteria_1_28-04-2023.2.zip)

The technical solutions built should be capable of building basic tests like these and be capable of answering some basic questions like these:

**_Use case 1._**

1)	Is an EU bond with 4 years remaining maturity eligible? If so, what are applicable haircuts?
2)	Are JGBs with a 3 year remaining maturity eligible? If so, what are applicable haircuts?
3)	Is GBP (cash) eligible? If so, what are applicable haircuts?
Using the following files for the first use case:
                           EU 1-5 2%
                           EU 1-5 15%
                           US 1-5 2%
                           US 1-5 4% Corp
                           US 1-5 4%
                            US Cash
These objects together specify the collateral eligible to be posted or received in this scenario.

**_Use case 2._**

Compare schedule 1 to schedule 2 and show the variances
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ISloyan/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ISloyan/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Schedule 1 | Schedule 2
-- | --
EU 1-5 2% | EU 1-5 2%
EU 1-5 15% | EU 1-5 15%
US 1-5 2% | US 1-5 2%
US 1-5 4% | US 1-5 4% Corp
US Cash | US 1-5 10%
  | US Cash

</body>

</html>

The criteria objects in the files need to be put/linked together to create the 2 schedules for you to use in use case 2.

## Useful apps, systems, projects, technologies, libraries and datasets
- [CDM](https://github.com/finos/common-domain-model)
- CDM object builder which was used to create the criteria objects (https://cdm-object-builder.dev.rosetta-technology.io/)
- CDM Eligible Criteria JSON objects  
[EligibleCollateralCriteria_1_28-04-2023 2.zip](https://github.com/finos/community/files/11383294/EligibleCollateralCriteria_1_28-04-2023.2.zip)
(same as attached zip above)
- Video explaining how to download and build the CDM (https://resources.finos.org/content/download-and-build-the-cdm/?projects-sigs=common-domain-model-cdm
-

**Comment by minesh-s-patel** [comment]
It would be great to embed the CDM Object Builder into FDC3, build a CDM collateral representation and show the difference in a human readable way.

**Comment by TheJuanAndOnly99** [comment]
Closing discussion in preparation for [this years hackathon](https://www.finos.org/hosted-events/2024-10-25-citi-india-hackathon-supported-by-finos). Feel free to reopen and update if its still relevant.
